SalesforceFoundation/EDA
# Prevent plan requirement having different program plan to the program plan of immediate plan requirement

# Critical Changes

# Changes
Prevent nested plan requirements from having different program plans.
# Issues Closed
Fixes #788 
# New Metadata
STG_CMP_ProgramPlans.cmp 
STG_CMP_ProgramPlansController.js
PREQ_UniqueProgramPlan_TDTM.cls
PREQ_UniqueProgramPlan_TEST.cls
Hierarchy_Settings__c.Unique_Program_Plan_Enforced__c
stgHelpUniqueProgramPlan
stgTabProgramPlan
stgUniqueProgramPlan
uniqueProgramPlanExp
# Deleted Metadata

# Testing Notes
1. Go to EDA Setting page and check "Enable Unique Program Plan Validation for Nested Plan Requirements" under program plan tab
2. Create a program plan: Test PP 1
3. Create a program plan: Test PP 2
4. Under Test PP 1 record page, create a plan requirement: Test PR 1 so that it is attached to Test PP 1
5. Under Test PR 1 record page, create a plan requirement: Test PR 2 so that it is attached to Test PR 1
6. Fill in the program plan as Test PP 2 and save and you should see an error message
7. Change it to Test PP 1 and make sure the record is saved
8. At Test PR 2 record page, edit the record and change the program plan to Test PP 2 and make sure seeing the error message again
9. Uncheck the setting and make sure the record page is saved even the program plan mismatches.
